morphir/runtime/src/org/finos/morphir/runtime/quick/EvaluatorQuick.scala

```diff
@@ -223,6 +227,9 @@ object EvaluatorQuick {
       case (Concept.List(elementConcept), Result.ListResult(elements)) =>
         val inners = elements.map(element => resultAndConceptToData(element, elementConcept))
         Data.List(inners, elementConcept)
+      case (Concept.Set(elementConcept), Result.SetResult(elements)) =>
+        val inners = elements.map(element => resultAndConceptToData(element, elementConcept))
+        Data.Set(mutable.Set.from(inners), elementConcept)
```

Comment:
Why is this mutable?

Reply:
I was looking at how Data.Map was implemented and it was using mutable.LinkedHashMap.  But you are right it should be immutable.  Will make the change.  Good catch!

Reply:
Oh, so Data.Map uses the mutable version for a deterministic order... I'm not sure what the equivalent for a set is. That said, we've had issues with losing dict ordering anyway, so we'll have to review all of that at some point later.

Reply:
Scala does have a [LinkedHashSet](https://www.scala-lang.org/api/2.13.x/scala/collection/mutable/LinkedHashSet.html) which is similar to a LinkedHashSet with the ordering.  But do we need to preserve the insertion order?

Reply:
To keep the order intact, you will probably need to keep the ordering in the MapResult by using LinkedHashMap there too otherwise you loose it there with a Map (HashMap)